examples/3.routes/8.tolls-and-ferries/interface.js

```diff
@@ -0,0 +1,260 @@
+    destination = data.legs[data.legs.length - 1].destination.properties.name;
+    // Go through the legs we receive from the backend - stations and final.
+    for (let i = 0; i < data.legs.length; i++) {
+      let ferrySteps = getNumberOfFerryStepsInLeg(data.legs[i]);
+      // Used to determine number of stops, at top of journey overview details
+      extraFerryStops += ferrySteps;
+      // 'Regular' route, no ferry. Add route to station as a leg
+      if (ferrySteps == 0) {
+        // Check for tolls
+        let tollsOnLeg = false;
+        if (containsTolls(data.legs[i])) {
+          tollsOnLeg = true;
+        }
+        // Station steps are split up into a drive to the station, followed by the charging info at the station
+        if (data.legs[i].type === 'station') {
+          journeyLegs.push({
+            tollsOnLeg,
+            rangeStartPercentage: data.legs[i].rangeStartPercentage,
+            rangeEndPercentage: data.legs[i].rangeEndPercentage,
+            type: 'pre-station',
+            duration: data.legs[i].duration,
+            distance: data.legs[i].distance,
+          });
+        }
+        journeyLegs.push({
+          ...data.legs[i],
+          tollsOnLeg,
+        });
+      }
+      // Leg contains ferry steps, so split it up
+      else {
+        splitLeg(data.legs[i]);
+      }
+    }
+  }
+
+  // the amount of stops in this route
+  const routeStops = `${data.charges + extraFerryStops ?? 0} stops`;
+
+  // A combined field containing several of the route meta data
+  document.getElementById('route-metadata').innerHTML = `${routeDistance} / ${routeStops} / ${routeEnergy}`;
+
+  // Populate journey overview
+  if (journeyLegs) {
+    // Show ferrys & tolls indicator if present on route
+    document.getElementById('toll').innerHTML = `${displayFerriesAndTollsIndicator()}`;
+    // Show start point
+    document.getElementById('journey-timeline').innerHTML += `
+    <div class="timeline-item timelineStartPoint start">
+    <div class='icon timeline-icon timelineStartPoint start'></div>
+    <div class='timeline-content'><p class='body-bold'>${origin}</p></div></div>`;
+    // Go through journey legs, displaying each one in order
+    for (let i = 0; i < journeyLegs.length; i++) {
+      let instructions = '';
+      let borderCrossingHTML = '';
+      switch (journeyLegs[i].type) {
+        case 'ferry':
+          instructions = `
+          <div class='timeline-item ferry'><div class="icon timeline-icon ferry"></div>
+          <div class='timeline-content'><p class='title'>
+          Ferry for ${(journeyLegs[i].distance / 1000).toFixed(1)} km</p></div>
+          <div><p class='body-small metal'>Approx ${getDurationString(journeyLegs[i].duration)}</p></div></div>
+          `;
+          break;
+        case 'station':
+          instructions = `
+          <div class='timeline-item station'>
+          <div class="icon timeline-icon chargeTurbo station"></div><div class="timeline-content">
```

Comment:
To keep it clean please move as much HTML/css to the HTML file and choose here if you need to render it.

Reply:
this was done in order to keep this example dynamic. so it's possible to change the query route and it will update the journey. but if you think it's better to statically generate (for simplicity), I can remove this logic

Reply:
Not fully agree with this. Whole car list is also build like this. Do agree that the whole timeline can be parsed in the JS file and not moved to HTML.